algo/src/main/java/org/neo4j/graphalgo/similarity/SimilarityProc.java

```diff
@@ -75,7 +74,12 @@ Long getDegreeCutoff(ProcedureConfiguration configuration) {
         return configuration.get("degreeCutoff", 0L);
     }
 
-    Stream<SimilaritySummaryResult> writeAndAggregateResults(Stream<SimilarityResult> stream, int length, boolean write, String writeRelationshipType, String writeProperty) {
+    Long getWriteBatchSize(ProcedureConfiguration configuration) {
+        return configuration.get("writeBatchSize", 1000L);
```

Comment:
1000 seems on the very low end esp. if we write sequentially, I'd up it to at least 10k or 100k